src/app/ui/AppProvider/NavigationProvider.tsx

```diff
@@ -15,10 +21,33 @@ const theme = {
   },
 };
 
+const linking: LinkingOptions<ReactNavigation.RootParamList> = {
+  prefixes: [DEEP_LINK_PREFIX!],
+  getStateFromPath: (path, options) => {
+    const state = getStateFromPath(path, options);
```

Comment:
Please make sure that parameters (e.g. email) are parsed correctly. Sometimes they don't if a parameter contains special characters. Wrap in `decodeURIComponent` if needed

Reply:
Since this is coming from a generated email, and the BE already encodes the query params I made the assumption this would work fine as this is how it's done on the other FE apps. Do you have an example that fails the parsing?

Reply:
To be honest, I don't have any specific examples of something that can go wrong from the UI side. However, I've got plenty of examples when the backend broke the parsing on their side which resulted in breaking of deep linking. Better safe than sorry.

Reply:
Sounds good. I'll add it in

Reply:
Ok I've done some testing around this and I can't get this to crash with any special characters. However, `decodeURIComponent` does crash if the argument isn't properly URL encoded. I'm willing to merge this in as-is, as trying to handle special characters is leading to more errors. What do you think?